hazelcast/statistics.py

```diff
@@ -0,0 +1,251 @@
+    _STAT_SEPARATOR = ","
+    _KEY_VALUE_SEPARATOR = "="
+    _EMPTY_STAT_VALUE = ""
+
+    _EXCEPTION_STR = "Could not collect data for the {}. It won't be collected again. {}"
+
+    logger = logging.getLogger("Statistics")
+
+    def __init__(self, client):
+        self._client = client
+        self._enabled = client.properties.get_bool(ClientProperties.STATISTICS_ENABLED)
+        self._cached_owner_address = None
+        self._statistics_timer = None
+        self._failed_gauges = set()
+
+    def start(self):
+        if not self._enabled:
+            return
+
+        period = self._client.properties.get_seconds(ClientProperties.STATISTICS_PERIOD_SECONDS)
+        if period <= 0:
+            default_period = self._client.properties.get_seconds_positive_or_default(
+                ClientProperties.STATISTICS_PERIOD_SECONDS)
+
+            self.logger.warning("Provided client statistics {} cannot be less than or equal to 0."
+                                "You provided {} as the configuration. Client will use the default value "
+                                "{} instead.".format(ClientProperties.STATISTICS_PERIOD_SECONDS.name,
+                                                     period,
+                                                     default_period))
+            period = default_period
+
+        def _statistics_task():
+            self._send_statistics()
+            self._statistics_timer = self._client.reactor.add_timer(period, _statistics_task)
+
+        self._statistics_timer = self._client.reactor.add_timer(period, _statistics_task)
+
+        self.logger.info("Client statistics enabled with the period of {} seconds.".format(period))
+
+    def shutdown(self):
+        if self._statistics_timer:
+            self._statistics_timer.cancel()
+
+    def _send_statistics(self):
+        owner_connection = self._get_owner_connection()
+        if owner_connection is None:
+            self.logger.debug("Cannot send client statistics to the server. No owner connection.")
+            return
+
+        stats = []
+        self._fill_metrics(stats, owner_connection)
+        self._add_near_cache_stats(stats)
+        self._add_runtime_and_os_stats(stats)
+        self._send_stats_to_owner("".join(stats), owner_connection)
+
+    def _send_stats_to_owner(self, stats, owner_connection):
+        request = client_statistics_codec.encode_request(stats)
+        self._client.invoker.invoke_on_connection(request, owner_connection)
+
+    def _add_runtime_and_os_stats(self, stats):
+        os_and_runtime_stats = self._get_os_and_runtime_stats()
+        for stat in os_and_runtime_stats:
+            self._add_stat(stats, stat, os_and_runtime_stats[stat])
+
+    def _get_os_and_runtime_stats(self):
+        psutil_stats = {}
+        if PSUTIL_ENABLED:
+            try:
+                # Availability: All platforms. Try to be sure that it is working.
+                if "phy_mem" not in self._failed_gauges:
```

Comment:
instead of writing the comment above, you can have a function such as:
```
def _can_collect_stat(name):
     return name not in self._failed_gauges
```
and here after removing the comment you can just have
 ```
 if self._can_collect_stat("phy_mem"):
```

and collecting the stat for each of this should be a separate function in my opinion. I believe the readability would be better if the layout of this function is as follows:

```
if self._can_collect_stat(statName1):
   self._collect_stat_name1()
if self._can_collect_stat(statName2):
   self._collect_stat_name2()
..
..
..
```

where `_collect_stat_name1()` will have the try catch block for the relevant stat.